.github/dependabot.yml

```diff
@@ -5,3 +5,16 @@ updates:
     directory: "/"
     schedule:
       interval: "weekly"
+    labels:
+      - "github-actions"
+      - "dependencies"
+    reviewers:
+      - "matthewfeickert"
+
+  # Ignore all pip dependencies to avoid PRs to update tests/constraints.txt
+  - package-ecosystem: "pip"
+    directory: "/"
+    schedule:
+      interval: "weekly"
+    ignore:
+      - dependency-name: "*"
```

Comment:
We don't want things like PR #2008 to ever happen, as `tests/constraints.txt` is something that should only ever be updated when the lower bounds in `setup.cfg`/`setup.py` need to be updated. I _think_ based on https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#ignore that this will ignore anything that is related to what Dependabot views as being in the Python ecosystem, but I'm not 100% sure on this. So if someone with more Dependabot experience can review this that would be great.